gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -119,12 +738,1491 @@ gum_stalker_add_call_probe (GumStalker * self,
+        g_print ("gum_exec_ctx_obtain_block_for - block is not full\n");
+        g_print ("(requirements & GUM_REQUIRE_RELOCATION) != 0\n");
+        g_print ("rl->eob = FALSE");
+#endif
+        rl->eob = FALSE;
+      }
+      else
+      {
+        break;
+      }
+    }
+    else if (gum_arm64_relocator_eob (rl))
+    {
+      break;
+    }
+  }
+
+  if (gc.continuation_real_address != NULL)
+  {
+    GumBranchTarget continue_target = { 0, };
+
+    continue_target.is_indirect = FALSE;
+    continue_target.absolute_address = gc.continuation_real_address;
+#if DEBUG_LEVEL == 10
+    g_print (
+        "gum_exec_ctx_obtain_block_for - gc.continuation_real_address != NULL\n");
+    g_print ("continue_target.absolute_address: %p",
+        continue_target.absolute_address);
+#endif
+    gum_exec_block_write_jmp_transfer_code (block, &continue_target, &gc);
+  }
+
+  gum_arm64_writer_put_brk_imm (cw, 14);
+
+  gum_arm64_writer_flush (cw);
+
+  block->code_end = (guint8 *) gum_arm64_writer_cur (cw);
+
+  block->real_begin = (guint8 *) rl->input_start;
+  block->real_end = (guint8 *) rl->input_cur;
+
+  gum_exec_block_commit (block);
+
+#if DEBUG_LEVEL == 10
+  g_print ("gum_exec_ctx_obtain_block_for - exit\n");
+  g_print ("\tcode_address_ptr: %p\n", code_address_ptr);
+  g_print ("\tcode_address: %p\n", *code_address_ptr);
+  g_print ("\tblock->code_begin: %p\n", block->code_begin);
+  g_print ("\tblock->code_end: %p\n", block->code_end);
+  g_print ("\tblock->real_begin: %p\n", block->real_begin);
+  g_print ("\tblock->real_end: %p\n", block->real_end);
+#endif
+  return block;
+}
+
+static void
+gum_exec_ctx_write_prolog (GumExecCtx * ctx,
+                           GumPrologType type,
+                           gpointer ip,
+                           GumArm64Writer * cw)
+{
+#if DEBUG_LEVEL == 10
+  g_print ("+ gum_exec_ctx_write_prolog - type: %d\n",type);
+#endif
+
+  /* 1) move to the red-zone */
+  gum_arm64_writer_put_sub_reg_reg_imm (cw, ARM64_REG_SP, ARM64_REG_SP,
+      GUM_RED_ZONE_SIZE);
+
+  /* 2) push registers that are going to be clobbered */
+  int immediate_for_sp = GUM_RED_ZONE_SIZE;
+  if (type == GUM_PROLOG_MINIMAL)
+  {
+    /* save the register used by stalker's code */
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X0, ARM64_REG_X1);
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X2, ARM64_REG_X3);
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X4, ARM64_REG_X5);
+
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_STALKER_REG_CTX,
+        ARM64_REG_X13);
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X14, ARM64_REG_X15);
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X16, ARM64_REG_X17);
+
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X29, ARM64_REG_X30);
+
+    gum_arm64_writer_put_instruction (cw, 0xD53B420F); /* MRS X15, NZCV */
+    gum_arm64_writer_put_push_reg_reg (cw, ARM64_REG_X30, ARM64_REG_X15);
+
+    /* 8 push of 16 */
+    immediate_for_sp += 16 * 8;
```

Comment:
The convention is: `quantity * size`, so this should be `8 * 16` for consistency.